File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/governance/GovernanceIdxCacheFromDbTest.java
package opensdk.sdk.apis.governance;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.GovernanceIdxCacheFromDbResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Governance RPC Test")
public class GovernanceIdxCacheFromDbTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC governance_idxCacheFromDb")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        GovernanceIdxCacheFromDbResponse response = sdk.governance.idxCacheFromDb(0).send();
        response.getResult();
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/governance/GovernanceIdxCacheFromDb.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../constant");

const sdk = new OpenSdk.GovernanceApi(new OpenSdk.ApiClient(RPC));

describe('governance_idxCacheFromDb API', () => {
    test('should return governance_idxCacheFromDb', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined();
            expect(Array.isArray(data) && data.length > 0).toBe(true);
            done();
        };

        sdk.idxCacheFromDb({}, callbackOne);
    });
});

